selfservice/flow/settings/hook.go

```diff
@@ -40,7 +40,7 @@ type (
 	PostHookPrePersistExecutorFunc func(w http.ResponseWriter, r *http.Request, a *Flow, s *identity.Identity) error
 
 	PostHookPostPersistExecutor interface {
-		ExecuteSettingsPostPersistHook(w http.ResponseWriter, r *http.Request, a *Flow, s *identity.Identity) error
+		ExecuteSettingsPostPersistHook(w http.ResponseWriter, r *http.Request, a *Flow, s *identity.Identity, continueWith *flow.ContinueWithContainer) error
```

Comment:
I agree, but we are fetching the flow after all the hooks have executed again, [here](https://github.com/ory/kratos/blob/1da0e40b6d83f4b39fc61308d93ec92cd33a7787/selfservice/flow/settings/hook.go#L284-L287) and return that. Since the `continue_with` items can contain tokens in plaintext, I think we don't want to store this in the DB. There is probably a good reason, that we re-fetch the flow at the end, so changing that could be breaking something non-obvious... WDYT?

Reply:
Replaced with the hook itself, and in the settings case, I am now carrying it over from the old flow.